plugins/woocommerce/includes/class-wc-install.php

```diff
@@ -863,7 +870,12 @@ private static function update() {
 	 * @param string|null $version New WooCommerce DB version or null.
 	 */
 	public static function update_db_version( $version = null ) {
-		update_option( 'woocommerce_db_version', is_null( $version ) ? WC()->version : $version );
+		if ( is_null( $version ) ) {
+			$last_db_version = array_key_last( self::$db_updates );
```

Comment:
Did exploration testing on the versions handling in https://3v4l.org/1GjX7: works as intended.